framework/source/class/qx/ui/window/Window.js

```diff
@@ -618,12 +732,15 @@ qx.Class.define("qx.ui.window.Window",
     */
 
     /**
-     * Closes the current window instance.
-     * Technically calls the {@link qx.ui.core.Widget#hide} method.
+     * Close the current window instance.
+     *
+     * Simply calls the {@link qx.ui.core.Widget#hide} method if the
+     * {@link qx.ui.win.Window#autoDestroy} property is false; otherwise 
+     * removes and destroys the window.
      */
     close : function()
     {
-      if (!this.isVisible()) {
+      if (!this.getAutoDestroy() && !this.isVisible()) {
```

Comment:
The original code just looked at isVisible because hide() and close() were essentially the same thing. If it was already not visible, through either a hide() call or a prior close() call, then there was nothing to do here. With autoDestroy true, we don't want to return. When close() is called, we definitely want it to close and be destroyed even if hide() had previously been called.